nibabel/streamlines/trk.py

```diff
@@ -395,8 +396,7 @@ def save(self, fileobj):
             pointing to TRK file (and ready to write from the beginning
             of the TRK header data).
         """
-        # Enforce little-endian byte order for header
-        header = self._default_structarr().newbyteorder('<')
+        header = self._default_structarr()
```

Comment:
I know it ends up the same, but maybe cleaner keeping the endian change in the save function with

```
self._default_structarr().newbyteorder('<').byteswap()
```

Reply:
Wouldn't that fail on little endian systems?

Reply:
Sorry - yes.  Then maybe, for clarity, add an `endianness=None` kwarg to `_default_structarr`, and pass `<` from `save`.

Reply:
That works. I followed the template in `WrapStruct.default_structarr`.

Let me know if you have any further comments.